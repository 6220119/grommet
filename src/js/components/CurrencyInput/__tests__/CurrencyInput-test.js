import React from 'react';
import 'jest-styled-components';
import 'regenerator-runtime/runtime';
import { render } from '@testing-library/react';

import { CurrencyInput } from '..';

describe('CurrencyInput', () => {
  test('basic', () => {
    const { container } = render(<CurrencyInput value={40.55} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test('brl in usa', () => {
    const { container } = render(
      <CurrencyInput locale="en-US" currency="BRL" value={40.55} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  test('brl in brazil', () => {
    const { container } = render(
      <CurrencyInput locale="pt-BR" currency="BRL" value={40.55} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  test('brl in brazil with custom currency display', () => {
    const { container } = render(
      <CurrencyInput
        locale="pt-BR"
        currency="BRL"
        value={40.55}
        numberFormatOptions={{ currencyDisplay: 'code' }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
