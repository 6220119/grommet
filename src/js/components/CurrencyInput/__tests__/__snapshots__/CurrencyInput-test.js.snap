// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyInput basic 1`] = `
.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c1:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1:focus > circle,
.c1:focus > ellipse,
.c1:focus > line,
.c1:focus > path,
.c1:focus > polygon,
.c1:focus > polyline,
.c1:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1:focus::-moz-focus-inner {
  border: 0;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0"
>
  <input
    autocomplete="off"
    class="c1"
    value="$40.55"
  />
</div>
`;

exports[`CurrencyInput brl in brazil 1`] = `
.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c1:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1:focus > circle,
.c1:focus > ellipse,
.c1:focus > line,
.c1:focus > path,
.c1:focus > polygon,
.c1:focus > polyline,
.c1:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1:focus::-moz-focus-inner {
  border: 0;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0"
>
  <input
    autocomplete="off"
    class="c1"
    value="R$ 40,55"
  />
</div>
`;

exports[`CurrencyInput brl in brazil with custom currency display 1`] = `
.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c1:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1:focus > circle,
.c1:focus > ellipse,
.c1:focus > line,
.c1:focus > path,
.c1:focus > polygon,
.c1:focus > polyline,
.c1:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1:focus::-moz-focus-inner {
  border: 0;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0"
>
  <input
    autocomplete="off"
    class="c1"
    value="BRL 40,55"
  />
</div>
`;

exports[`CurrencyInput brl in usa 1`] = `
.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
}

.c1:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1:focus > circle,
.c1:focus > ellipse,
.c1:focus > line,
.c1:focus > path,
.c1:focus > polygon,
.c1:focus > polyline,
.c1:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c1:focus::-moz-focus-inner {
  border: 0;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:-moz-placeholder,
.c1::-moz-placeholder {
  opacity: 1;
}

.c0 {
  position: relative;
  width: 100%;
}

<div
  class="c0"
>
  <input
    autocomplete="off"
    class="c1"
    value="R$40.55"
  />
</div>
`;
